### ISSUE TITLE
Is it possible to use Voyager user info outside the Voyager admin?

### ISSUE THREAD

**Issue by JanisE** [thedevdojo/voyager#2902]
- Laravel Version: 5.5.39
- Voyager Version: 1.0.16
- PHP Version: 7.0.28
- Database Driver & Version: mysql, MariaDB 10.0.34

### Description:

Is it possible to use Voyager user info outside the Voyager admin?

For example, I have no authentication on my public site, but I would like to (silently) check whether the particular user is logged in the admin (using the admin login form) and based on their permissions do something different on the public site.

Currently, `Auth::user()` returns `null` in a controller of the public part.

**Comment by fletch3555** [comment]
Authentication is session based, so the parts of your site will need to share a session.  If you have voyager on a subdomain or something, you'll have to change the session configuration to work with that.  If you just have Voyager installed normally, then it should work fine.  However, if a session does not include a valid authentication for Voyager, then `Auth::user()` will definitely return null.  Additionally, if you've changed the auth config at all, it may not work.

In short, yes, it should work fine, so long as you configure it in a way that will let it work.  Without being able to see your configs and code, I can't tell you why it's not working.  And github issues aren't intended for troubleshooting your site in this way.  If you need further help, please feel free to ask on our Slack group and we'll try to help figure it out.

**Comment by JanisE** [comment]
With the help at https://laravel-voyager.slack.com I determined that in my case the problem was only with the controllers called by the API routes, because the API middleware does not include session management which Voyager authentication depends on.

So, the solution for me was to create an api-with-session middleware which combines api and web middleware `VerifyCsrfToken` removed (see this as an example: https://stackoverflow.com/a/39605396/99904).
